explanation/journey-of-a-change-to-production.rst

```diff
@@ -0,0 +1,317 @@
+-----------------------------------------------------------------------------------
+
+The same buildbot poller script takes care of periodically merging the latest
+changes in the ``stable`` branch to the ``db-devel`` branch. Then the changes
+get tested on buildbot before they get merged to the ``db-stable`` branch.
+
+As mentioned in the explanation of the database deployment process, the changes
+in the ``db-stable`` branch get submitted for merge to the ``master`` branch
+by a Launchpad developer after deploying a database patch. Once the merge
+proposal gets approved and merged, buildbot runs the test suite and if it
+passes, the changes then get merged to the ``stable`` branch.
+
+How does a Launchpad branch get deployed?
+-----------------------------------------
+
+When a change is merged to the ``master`` branch, there is a
+`launchpad-build-charm Jenkins job`_ that builds a Launchpad deployment tarball
+of the latest commit in that branch and publishes it to a well-known bucket
+on the ProdStack SWIFT storage. All the Launchpad charms use these tarballs
+to deploy the Launchpad source and its dependencies.
+
+Similarly, there is a `launchpad-build-db-charm Jenkins job`_ that builds a
+tarball of the latest commit in the ``db-devel`` branch and publishes it to
+SWIFT.
+
+In the production environment, all the Launchpad units except the
+``launchpad-db-update`` unit run the tarball of a specified commit in the
+``stable`` branch. The ``launchpad-db-update`` unit runs the tarball of the
+latest commit in the ``db-stable`` branch because it is used to apply the
+database patches.
+
+In the ``staging`` environment, all the Launchpad units run the tarball of the
+latest commit in the ``db-stable`` branch and in the ``qastaging`` environment,
+all the Launchpad units run the tarball of the latest commit in the ``stable``
+branch. Due to this, database changes can be deployed to the qastaging
+environment only after they have been merged to the ``master`` after the
+production deployment and promoted to the ``stable`` branch after a successful
+buildbot run.
+
+The database changes have to be deployed in the ``qastaging`` environment
+manually by following a process similar to the production ``fastdowntime``
+deployment.
+
+.. _launchpad-build-charm Jenkins job: https://jenkins.ols.ps5.canonical.com/job/launchpad-build-charm/
+.. _launchpad-build-db-charm Jenkins job: https://jenkins.ols.ps5.canonical.com/job/launchpad-build-db-charm/
+
+How does the Launchpad deployment process work?
+-----------------------------------------------
+
+Launchpad is deployed to a Canonical ProdStack environment using Juju charms
+and ``mojo``. We use the ``lp`` spec in the `launchpad-mojo-specs repository`_
+to define the Juju bundle (see ``lp/bundle.yaml``) used to deploy the Launchpad
+stack.
+
+For deploying changes to the pre-production environments, we directly invoke
+the ``mojo run`` command from the ``stg-launchpad`` account on the Launchpad
```

Comment:
This might lead to confusion for a new-starter because one needs to source the env or do `in-model=qastaging` before the `mojo run`, so perhaps better to add that?